DependencyInjection/Configuration.php

```diff
@@ -368,34 +370,61 @@ private function addSharedPluginNodes(ArrayNodeDefinition $pluginNode, $disableA
                         ->cannotBeEmpty()
                     ->end()
                     ->arrayNode('config')
-                        ->addDefaultsIfNotSet()
```

Comment:
if you remove this, you MUST check whether the `config` key is defined in the DI extension before using it. Currently, the DI extension relies on the fact that the Configuration class ensures it is always defined, but your change means it is not the case anymore.